samples: USB mass storage sample.yaml cleanup

ajaykish opened the pull request:
Add console harness support for USB mass storage test.
Also, using overlay config for DISK_ACCESS configuration
and accessing the different config file with different
method.


Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>

finikorg reviewed (changes requested):
I wonder could we better use overlay config for `DISK_ACCESS` configuration, we could have 2 tests with different access method, check `samples/net/echo_server/sample.yaml`:

```
  test_usbnet:
    harness: net
    extra_args: OVERLAY_CONFIG="overlay-netusb.conf"
    platform_whitelist: quark_se_c1000_devboard 96b_carbon
    tags: net usb
```

jfischer-phytec-iot commented on samples/subsys/usb/mass/prj.conf:
```diff
@@ -1,6 +1,6 @@
 CONFIG_STDOUT_CONSOLE=y
 CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_DEFAULT_LEVEL=1
+CONFIG_SYS_LOG_DEFAULT_LEVEL=3
```
Please leave SYS_LOG_LEVEL as it was.

galak replied:
@ajaykish what does 'internal test automation' mean?  Why doesn't your internal test automation set this to the level it wants?

nashif replied:
> @jfischer-phytec-iot This is required for internal test automation.

This is not a good enough reason, this is a sample, not a test and making changes here does affect everyone using it. Depending on a certain log level in a sample might not provide a solid  test environment.

jfischer-phytec-iot commented:
>I wonder could we better use overlay config for DISK_ACCESS configuration, we could have 2 tests with different access method

+1

ajaykish committed:
samples: USB mass storage sample.yaml cleanup

Add console harness support for USB mass storage test.
Also, using overlay config for DISK_ACCESS configuration
and accessing the different config file with different
method.

Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

ajaykish commented on samples/subsys/usb/mass/overlay-flash-disk.conf:
```diff
@@ -0,0 +1,4 @@
+#config to disk access over SPI flash
```
Done.

ajaykish commented:
Thanks @finikorg , as per your suggestion added  the extra_configs in sample.yaml .

jfischer-no reviewed (commented):
LGTM, but maybe we should wait for #10168
